### PR TITLE
Enhance date and time validation logic

### LINKED ISSUE #108: Improve validation for date and time

### PR DESCRIPTION
**Pull request by tzuennn**
Fixes #108 
Previously, the isValidTime method only checked for the correct format of time strings but did not validate that the hours and minutes fell within acceptable ranges. This led to scenarios where invalid times (like "25:00" or "12:60") were incorrectly considered valid. When these incorrect dates are used, no error message is triggered, and nothing is added to the application.

Similar issues were encountered with validation of date. Examples like "2024-03-99, 2023-50-01" were all previously considered valid.

To address this issue, I have modified the isValidTime method to:
- Split the input string into hours and minutes.
- Validate that hours are between 0 and 23 and minutes are between 0 and 59.
- Explicitly throw a NullPointerException when the input is null

For the issue of date, instead of using regex for validation, I used LocalDate parser to check for valid date.

With these changes, the validation process is now more robust, preventing invalid time strings from being accepted.

### CODE REVIEW

File: src/main/java/seedu/address/model/event/Time.java [hunk 1]
```diff
@@ -40,9 +35,31 @@ public Time(String time) throws DateTimeParseException {
      * Returns true if a given string is a valid time.
      */
     public static boolean isValidTime(String test) {
```
Comment (Fluffykan) [inline src/main/java/seedu/address/model/event/Time.java:37]: Similar to Date, this test could be done with `LocalTime.parse(test)`, catching the `DateTimeParseException` in the event of an invalid time provided by the user.

**Review by Fluffykan — CHANGES_REQUESTED** [review]
Just need minor tweaks, but nice catch on the fact that it accepts invalid times 👍

File: src/main/java/seedu/address/logic/parser/volunteercommandparser/VolunteerNewCommandParser.java [hunk 2]
```diff
@@ -60,7 +60,7 @@ public VolunteerAddCommand parse(String args) throws ParseException {
         Time endTime = VolunteerParserUtil.parseTime(argMultimap.getValue(VOLUNTEER_PREFIX_END_TIME).get());
```
Comment (Fluffykan) [inline src/main/java/seedu/address/logic/parser/volunteercommandparser/VolunteerNewCommandParser.java:60]: Might want to ensure that start time is before end time here too, similar to Event

File: src/main/java/seedu/address/model/event/Time.java (hunk 1 above)
Comment (keerthigkaarthik) [reply src/main/java/seedu/address/model/event/Time.java:37]: The parse method accepts seconds (e.g. 10:15:15) and if we don't want to deal with that you can have a DateTimeParser.ofPattern("HH:mm").

File: src/main/java/seedu/address/model/volunteer/Time.java [hunk 3]
```diff
@@ -31,20 +28,51 @@ public class Time {
...
         requireNonNull(time);
         checkArgument(isValidTime(time), MESSAGE_CONSTRAINTS);
-        this.eventTime = LocalTime.parse(time);
+        this.volunteerTime = LocalTime.parse(time);
     }
 
     /**
      * Returns true if a given string is a valid time.
      */
     public static boolean isValidTime(String test) {
```
[the remark this answers is not in the document]
Comment (keerthigkaarthik) [reply src/main/java/seedu/address/model/volunteer/Time.java:37]: I thought we were removing volunteer time? Can we just comment or delete this for now?

**Review by keerthigkaarthik — COMMENTED** [review]
Great find. Ty!!

File: src/main/java/seedu/address/model/event/Time.java (hunk 1 above)
Comment (tzuennn) [reply src/main/java/seedu/address/model/event/Time.java:38]: Nice! did not think of that, alr fixed the issue to use local time parser for validation

File: src/main/java/seedu/address/model/volunteer/Time.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (tzuennn) [reply src/main/java/seedu/address/model/volunteer/Time.java]: Yup the recent commit removed the time class in volunteer

File: src/main/java/seedu/address/logic/parser/volunteercommandparser/VolunteerNewCommandParser.java (hunk 2 above)
Comment (tzuennn) [reply src/main/java/seedu/address/logic/parser/volunteercommandparser/VolunteerNewCommandParser.java]: as time is removed, this validation is no longer needed for volunteer. thanks for pointing out the issue tho!

**Review by Fluffykan — APPROVED** [review]
LGTM 👍

**Pull request merged**